ui/src/screen/project.ts

```diff
@@ -24,51 +26,43 @@ export const fetch = (projectUrn: Urn): void => {
 
   proxy.client.project
     .get(projectUrn)
-    .then(async current => {
+    .then(async prj => {
       const peers = await proxy.client.project.listPeers(projectUrn);
       const peerSelection = project.userList(peers);
       throwUnlessPeersPresent(peerSelection, projectUrn);
       screenStore.success({
         peers,
         peerSelection,
-        project: current,
-        requestInProgress: null,
+        project: prj,
         selectedPeer: peerSelection[0],
       });
     })
     .catch(err => screenStore.error(error.fromUnknown(err)));
 };
 
-export const refreshPeers = (): void => {
+export const refreshPeers = async (): Promise<void> => {
   const screen = get(screenStore);
 
   if (screen.status === remote.Status.Success) {
-    const { data: current } = screen;
-    const { requestInProgress } = current;
+    try {
+      const peers = await refreshExecutor.run(abort =>
+        proxy.client.project.listPeers(screen.data.project.urn, { abort })
+      );
+      if (peers === undefined) {
+        return;
+      }
 
-    if (requestInProgress) {
-      requestInProgress.abort();
+      const peerSelection = project.userList(peers);
+      throwUnlessPeersPresent(peerSelection, screen.data.project.urn);
+      screenStore.success({
+        ...screen.data,
+        peers,
+        peerSelection,
+      });
+    } catch (err) {
+      screenStore.error(error.fromUnknown(err));
+      return;
```

Comment:
Why do we need an explicit `return` here?

Reply:
We don’t need it. Fixed in https://github.com/radicle-dev/radicle-upstream/pull/1985/commits/e379e2e1e3d7f4bf69fd4293f14ffe076084e96c